### PR TITLE
Fixes gr.Markdown() does not render spaces around links correctly

### PR DESCRIPTION
**Pull request by dawoodkhan82**
## Description
Removed styling that created issues around any children of headers (links, bold, italics, text, etc.)

See `markdown_example` demo to view changes this causes. 
Closes: #7359

## 🎯 PRs Should Target Issues

Before your create a PR, please check to see if there is [an existing issue](https://github.com/gradio-app/gradio/issues) for this change. If not, please create an issue before you create this PR, unless the fix is very small. 

Not adhering to this guideline will result in the PR being closed. 

## Tests

1. PRs will only be merged if tests pass on CI. To run the tests locally, please set up [your Gradio environment locally](https://github.com/gradio-app/gradio/blob/main/CONTRIBUTING.md) and run the tests: `bash scripts/run_all_tests.sh`

2. You may need to run the linters: `bash scripts/format_backend.sh` and `bash scripts/format_frontend.sh`

### CODE REVIEW

File: js/markdown/shared/MarkdownCode.svelte [hunk 1]
```diff
@@ -160,14 +160,4 @@
 	span.md :global(.md-header-anchor > svg) {
 		color: var(--body-text-color-subdued);
 	}
-
-	span :global(h1),
```
Comment (dawoodkhan82) [inline js/markdown/shared/MarkdownCode.svelte:164]: @pngwn I'm curious why we needed this? It removes any whitespace from the children of headers. If we do need it I could also add margins to those html elements separately.

Comment (pngwn) [reply js/markdown/shared/MarkdownCode.svelte:164]: I think it is for the header links + anchors that are generated with certain settings enabled.

**Comment by abidlabs** [comment]
Ah I see the issue that @pngwn is referring to. It happens when you set `header_links=True`. e.g.

```py
import gradio as gr

with gr.Blocks() as demo:
    gr.Markdown("""
    # abc
    def
    """, header_links=True)
    demo.launch()
```

The links get rendered on a separate line now:

image

**Review by abidlabs — CHANGES_REQUESTED** [review]
See above

**Review by abidlabs — APPROVED** [review]
LGTM @dawoodkhan82! I added a story to prevent regressions

**Pull request merged**